### PR TITLE
fix: make executor threads deamon

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## This PR

Fixes https://github.com/open-feature/java-sdk/issues/600

Makes task executor threads daemons.

### CODE REVIEW

**Review by toddbaert — APPROVED** [review]
Great idea!

File: src/main/java/dev/openfeature/sdk/EventSupport.java [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package dev.openfeature.sdk;
 
+import edu.umd.cs.findbugs.annotations.Nullable;
```
Comment (toddbaert) [inline src/main/java/dev/openfeature/sdk/EventSupport.java:3]: I think this is the wrong import... not your change but can you fix it?

Comment (Kavindu-Dodan) [reply src/main/java/dev/openfeature/sdk/EventSupport.java:3]: yup, will fix that to have consistency across other usages

Comment (Kavindu-Dodan) [reply src/main/java/dev/openfeature/sdk/EventSupport.java]: done :)

**Review by toddbaert — APPROVED** [review]

**Pull request merged**